mirrord-layer/src/error.rs

```diff
@@ -53,3 +53,31 @@ pub enum LayerError {
     #[error("mirrord-layer: Failed inserting listen, already exists!")]
     ListenAlreadyExists,
 }
+
+//Todo: https://stackoverflow.com/questions/39150216/implementing-a-trait-for-multiple-types-at-once
+
+// what should the mapping be between LayerError and integer types
+
+// mapping based on - https://man7.org/linux/man-pages/man3/errno.3.html
+
+impl From<LayerError> for i32 {
+    fn from(error: LayerError) -> Self {
+        match error {
```

Comment:
Log the error with `error!` before the `match` so we get a bit more context.

Reply:
Does that mean we remove the log message from the hook? that would make the `map_err` ignore the argument in the closure